File: token-client-core/src/main/java/no/nav/security/token/support/client/core/oauth2/ClientCredentialsTokenClient.java
package no.nav.security.token.support.client.core.oauth2;

import no.nav.security.token.support.client.core.http.OAuth2HttpClient;

import java.util.Map;

public class ClientCredentialsTokenClient extends AbstractOAuth2TokenClient<ClientCredentialsGrantRequest> {

    public ClientCredentialsTokenClient(OAuth2HttpClient oAuth2HttpClient) {
        super(oAuth2HttpClient);
    }

    @Override
    protected Map<String, String> formParameters(ClientCredentialsGrantRequest grantRequest) {
        return Map.of();
    }
}

File: token-client-core/src/main/kotlin/no/nav/security/token/support/client/core/oauth2/ClientCredentialsTokenClient.kt
package no.nav.security.token.support.client.core.oauth2

import no.nav.security.token.support.client.core.OAuth2ParameterNames.SCOPE
import no.nav.security.token.support.client.core.http.OAuth2HttpClient

class ClientCredentialsTokenClient(oAuth2HttpClient : OAuth2HttpClient) : AbstractOAuth2TokenClient<ClientCredentialsGrantRequest>(oAuth2HttpClient) {

    override fun formParameters(grantRequest : ClientCredentialsGrantRequest) = LinkedHashMap<String, String>().apply {
        put(SCOPE, grantRequest.clientProperties.scope.joinToString(" "))
    }
}